### ISSUE TITLE
Auto discover responses

### ISSUE THREAD

**Issue by DenuxPlays** [juhaku/utoipa#1094]
Hey,

just like with request_body and response schemas whole responses (with traits `ToResponse` and `IntoResponses`) do not get auto discovered when used in the paths macros.

**Comment by juhaku** [comment]
Yup, that's true, as indicated here https://github.com/juhaku/utoipa/issues/1065 there is not yet any implementation for them to be automatically recognized. Not sure whether I will do it to the first version of 5.0.0 and there might be some fundamental changes to be made for those traits as I am not very convinced that they perform the best of their intended use case.

**Comment by DenuxPlays** [comment]
Okay great to hear.
I agree I found it very confusing working the first time with ToResponse and IntoResponses but not sure if I can speak for the whole community.

**Comment by juhaku** [comment]
Yeah, I'm kind of tempted to deprecate them and even perhaps to remove them :sweat_smile: But there are people using them so not really viable option as of yet. I kind of want to find a way to leverage the `ToSchema` as a base and then coating that with additional data for making it a "response" what can be referenced and what not later on. Now the `ToResponse` and `IntoResponses` does whole bunch of needless processing and adds to the complexity too much maintenance overhead.

**Comment by DenuxPlays** [comment]
I guess deprecating it would be a good idea if a better solution can be found.
I would be fine with removal as this is a breaking change release but again I cannot speak for the community.
I think leveraging the `ToSchema` trait would be a good idea.

**Comment by juhaku** [comment]
Related issues #1025  #465
